signature-aggregator/aggregator/aggregator_test.go

```diff
@@ -85,3 +153,248 @@ func TestCreateSignedMessageFailsWithoutSufficientConnectedStake(t *testing.T) {
 		"failed to connect to a threshold of stake",
 	)
 }
+
+func makeAppRequests(
+	chainID ids.ID,
+	requestID uint32,
+	connectedValidators *peers.ConnectedCanonicalValidators,
+) []ids.RequestID {
+	var appRequests []ids.RequestID
+	for _, validator := range connectedValidators.ValidatorSet {
+		for _, nodeID := range validator.NodeIDs {
+			appRequests = append(
+				appRequests,
+				ids.RequestID{
+					NodeID:             nodeID,
+					SourceChainID:      chainID,
+					DestinationChainID: chainID,
+					RequestID:          requestID,
+					Op: byte(
+						message.AppResponseOp,
+					),
+				},
+			)
+		}
+	}
+	return appRequests
+}
+
+func TestCreateSignedMessageRetriesAndFailsWithoutP2PResponses(t *testing.T) {
+	aggregator, mockNetwork := instantiateAggregator(t)
+
+	var (
+		connectedValidators, _ = makeConnectedValidators(2)
+		requestID              = aggregator.currentRequestID.Load() + 1
+	)
+
+	chainID, err := ids.ToID(utils.RandomBytes(32))
+	if err != nil {
+		panic(err)
+	}
+
+	msg, err := warp.NewUnsignedMessage(0, chainID, []byte{})
+	require.Equal(t, nil, err)
+
+	subnetID, err := ids.ToID(utils.RandomBytes(32))
+	require.Equal(t, nil, err)
+	mockNetwork.EXPECT().GetSubnetID(chainID).Return(
+		subnetID,
+		nil,
+	)
+
+	mockNetwork.EXPECT().ConnectToCanonicalValidators(subnetID).Return(
+		connectedValidators,
+		nil,
+	)
+
+	appRequests := makeAppRequests(chainID, requestID, connectedValidators)
+	for _, appRequest := range appRequests {
+		mockNetwork.EXPECT().RegisterAppRequest(appRequest).Times(
+			maxRelayerQueryAttempts,
+		)
+	}
+
+	mockNetwork.EXPECT().RegisterRequestID(
+		requestID,
+		len(appRequests),
+	).Return(
+		make(chan message.InboundMessage, len(appRequests)),
+	).Times(maxRelayerQueryAttempts)
+
+	var nodeIDs set.Set[ids.NodeID]
+	for _, appRequest := range appRequests {
+		nodeIDs.Add(appRequest.NodeID)
+	}
+	mockNetwork.EXPECT().Send(
+		gomock.Any(),
+		nodeIDs,
+		subnetID,
+		subnets.NoOpAllower,
+	).Times(maxRelayerQueryAttempts)
+
+	_, err = aggregator.CreateSignedMessage(msg, subnetID, 80)
+	require.ErrorContains(
+		t,
+		err,
+		"failed to collect a threshold of signatures",
+	)
+}
+
+func TestCreateSignedMessageSucceeds(t *testing.T) {
+	var msg *warp.UnsignedMessage // to be signed
+	chainID, err := ids.ToID(utils.RandomBytes(32))
+	if err != nil {
```

Comment:
Let's use `require.NoError` rather than panicking.

Reply:
this isn't necessary now that we're using `GenerateTestID`, since it doesn't return an error